tools/codemods/src/codemods/popover-v12/transform.ts

```diff
@@ -25,85 +62,266 @@ const propNamesToRemove = [
-    const elements = source.findJSXElements(componentName);
+    /**
+     * Look for named import declarations for the given package name.
+     */
+    const namedExportToLookFor = lgPackageComponentsMap[packageName];
+    matchingImportDeclaration.find(j.ImportSpecifier).forEach(path => {
+      /**
+       * If the named import does not match the component to look for, keep checking the next import.
+       */
+      const matchesComponentImport =
+        path.node.imported.name === namedExportToLookFor;
 
-    if (elements.length === 0) return;
+      if (!matchesComponentImport) {
+        return;
+      }
 
-    elements.forEach(element => {
-      addJSXAttributes({
-        j,
-        element,
-        propName: 'usePortal',
-        propValue: true,
-        commentOverride: `${MIGRATOR_ERROR.manualAdd} prop: renderMode`,
+      /**
+       * If the named import is aliased, use the alias name.
+       */
+      componentsToTransform.push(
+        path.node.local ? path.node.local.name : namedExportToLookFor,
+      );
+    });
+
+    return componentsToTransform;
+  };
+
+  /**
+   * By default, transform all components in the default packages. If the `packages` option is provided,
+   * only transform the provided packages.
+   */
+  const packagesToCheck = providedPackages || defaultPackages;
+  const packagesForPropConsolidation = packagesToCheck.filter(
+    packageName => lgPackageComponentForPropConsolidationMap[packageName],
+  );
+  const packagesForPropRemoval = packagesToCheck.filter(
+    packageName => lgPackageComponentForPropRemovalMap[packageName],
+  );
+  const packagesForPropReplacement = packagesToCheck.filter(
+    packageName => lgPackageComponentForPropReplacementMap[packageName],
+  );
+
+  /**
+   * This block handles transforming components that require prop consolidation.
+   */
+  packagesForPropConsolidation.forEach(packageName => {
+    const componentsForPropConsolidation = getAllComponentsToTransform({
+      packageName,
+      lgPackageComponentsMap: lgPackageComponentForPropConsolidationMap,
+    });
+
+    componentsForPropConsolidation.forEach(componentName => {
+      const elements = source.findJSXElements(componentName);
+
+      if (elements.length === 0) return;
+
+      elements.forEach(element => {
+        const attributes = getJSXAttributes(j, element, 'renderMode');
+
+        if (attributes.length === 0) {
+          addJSXAttributes({
+            j,
+            element,
+            propName: 'usePortal',
+            propValue: true,
+            commentOverride: `${MIGRATOR_ERROR.manualAdd} prop: renderMode`,
+          });
+        }
+
+        consolidateJSXAttributes({
+          j,
+          element,
+          propToRemove: 'usePortal',
+          propToUpdate: 'renderMode',
+          propMapping: {
+            false: 'inline',
+            true: 'portal',
+          },
+          propToRemoveType: 'boolean',
+        });
```

Comment:
might be worth extracting these into their own functions. That would make this transform file easier to read